code/modules/transport/pods/MainWeapon.dm

```diff
@@ -380,6 +380,608 @@
+					src.sadbuzz()
+					return
+
+				if(ship.bound_height == 64)
+					src.do_construct(mode,TRUE)
+				else
+					src.do_construct(mode,FALSE)
+
+				playsound(src.loc, 'sound/machines/constructor_work.ogg', 30, 1)
+				// Necessary when not calling base fire proc
+				logTheThing(LOG_COMBAT, user, "driving [ship.name] fires [src.name], attempting to construct [mode == EFIF_MODE_WALLS ? "walls" : "floors"] at [log_loc(ship)].")
+
+			if(EFIF_MODE_REPAIR)
+				if(ON_COOLDOWN(src, "fire", firerate))
+					return
+				if(length(src.active_fields) >= 1)
+					return
+
+				if(ship.bound_height == 64)
+					src.do_repair(TRUE,user)
+				else
+					src.do_repair(FALSE,user)
+
+				// Necessary when not calling base fire proc
+				logTheThing(LOG_COMBAT, user, "driving [ship.name] fires [src.name], attempting to repair an area at [log_loc(ship)].")
+
+			else
+				..()
+
+	proc/sadbuzz()
+		for (var/mob/M in ship)
+			M.playsound_local(ship, 'sound/machines/buzz-two.ogg', 5, TRUE, ignore_flag = SOUND_IGNORE_SPACE)
+
+	///Helper proc that checks if enough sheets are available for building given current build mode and ship size
+	proc/check_sheets(var/mode_specify)
+		if(!mode_specify)
+			mode_specify = mode
+		. = FALSE
+		var/sizemult = 1
+		if(ship.bound_height == 64)
+			sizemult = 2
+		if(src.wide_field)
+			sizemult += 2
+		if(mode_specify == EFIF_MODE_FLOORS && src.steel_sheets >= EFIF_FLOOR_COST * sizemult)
+			return TRUE
+		else if(mode_specify == EFIF_MODE_R_FLOORS && src.steel_sheets >= EFIF_R_FLOOR_COST * sizemult)
+			return TRUE
+		else if(mode_specify == EFIF_MODE_WALLS && src.steel_sheets >= EFIF_WALL_COST * sizemult)
+			return TRUE
+
+	///Ship arms' proc to find and attempt loading of sheets on the ground
+	proc/scoop_up_sheets()
+		var/turf/load_from
+		var/loaded_stuff
+		if(ship.bound_height == 64) //always load at the front side of the pod, left offset
+			switch(ship.dir)
+				if(NORTH)
+					load_from = locate(ship.x,ship.y+2,ship.z)
+				if(EAST)
+					load_from = locate(ship.x+2,ship.y+1,ship.z)
+				if(SOUTH)
+					load_from = locate(ship.x+1,ship.y-1,ship.z)
+				if(WEST)
+					load_from = locate(ship.x-1,ship.y,ship.z)
+		else
+			load_from = get_step(ship,ship.dir)
+		if(!load_from)
+			return
+		for (var/obj/O in load_from)
+			if(src.sheet_load_helper(O) >= LOAD_SUCCESS)
+				loaded_stuff = TRUE
+		if(loaded_stuff)
+			for (var/mob/M in ship)
+				M.playsound_local(ship, 'sound/machines/chime.ogg', 5, TRUE, ignore_flag = SOUND_IGNORE_SPACE)
+			ship.visible_message("<b>[ship]</b> loads metal sheets into its tool.")
+
+	///Sheet-loading proc for both automatic and manual loading. Set up for flag-based return, for responsive player messaging.
+	///Returns FALSE if provided object is unsuitable, LOAD_FULL (1) if storage is full, and/or LOAD_SUCCESS (2) if sheets were loaded.
+	proc/sheet_load_helper(var/obj/O,var/mob/user)
```

Comment:
Consider just using sheet stacking behaviour rather than reimplementing it here

Reply:
Not entirely sure what you mean by "just using" - I took a peek at sheet stacking behavior and it has a lot of code not relevant to this functionality, unless I'm looking in the wrong spot, and I do have some return code stuff for load message formatting woven into the implementation here.

Reply:
I think you could just store a stack of sheets and call `stack_item` on them?

Reply:
Was a little more complex than that, but done.